### PR TITLE
fix #5760: transition async component error

### LINKED ISSUE #5760: out-in transition issue with async components
### Version
2.3.0

### Reproduction link
[https://jsfiddle.net/z11fe07p/1959/](https://jsfiddle.net/z11fe07p/1959/)

### Steps to reproduce
- load https://jsfiddle.net/z11fe07p/1959/
- click the "B" button

### What is expected?
The previous component (A) should disappear before component B appears

### What is actually happening?
Components A and B are visible simultaneously

---
This behavior is happens only when component B is not already loaded. Next time you switch from A to B, everything is fine.

### PR DESCRIPTION
**Pull request by gebilaoxiong**
issue: https://github.com/vuejs/vue/issues/5760

**What kind of change does this PR introduce?** (check at least one)

- [x] Bugfix
- [ ] Feature
- [ ] Code style update
- [ ] Refactor
- [ ] Build-related changes
- [ ] Other, please describe:

**Does this PR introduce a breaking change?** (check one)

- [ ] Yes
- [x] No

If yes, please describe the impact and migration path for existing applications:

**The PR fulfills these requirements:**

- [x] It's submitted to the `dev` branch for v2.x (or to a previous version branch), _not_ the `master` branch
- [x] When resolving a specific issue, it's referenced in the PR's title (e.g. `fix #xxx[,#xxx]`, where "xxx" is the issue number)
- [x] All tests are passing: https://github.com/vuejs/vue/blob/dev/.github/CONTRIBUTING.md#development-setup
- [x] New/updated tests are included

If adding a **new feature**, the PR's description includes:
- [ ] A convincing reason for adding this feature (to avoid wasting your time, it's best to open a suggestion issue first and wait for approval before working on it)

**Other information:**

### CODE REVIEW

**Comment by gebilaoxiong** [comment]
It 's too late today and going to work tomorrow

tomorrow I will fill the unit test ... 

sorry

File: src/platforms/web/runtime/components/transition.js [hunk 1]
```diff
@@ -72,6 +78,23 @@ function isSameChild (child: VNode, oldChild: VNode): boolean {
...
+function isAsyncCompPlaceholder (child: VNode): boolean {
+  if (isUndef(child)) {
+    return false
+  }
+  return child.isComment && isDef(child.asyncFactory)
+}
+
+function createAsyncPlaceholder (vnode: VNode): VNode {
+  const clone = cloneVNode(vnode)
+  clone.tag = 'div'
```
Comment (Kingwl) [inline src/platforms/web/runtime/components/transition.js]: IMO 'div' is not a good idea

Comment (gebilaoxiong) [inline src/platforms/web/runtime/components/transition.js]: But it is invisible

Comment (Kingwl) [inline src/platforms/web/runtime/components/transition.js]: ping @yyx990803

**Comment by gebilaoxiong** [comment]
@Kingwl  now, I use a transition placeholder... can you help me to review it : )

**Comment by yyx990803** [comment]
Thanks @gebilaoxiong for the hard work. I found out a simpler fix in https://github.com/vuejs/vue/commit/c3cdfcfa68126783307d74fca22b136cd41c474a but your PR helped a lot in identifying the problem.

**Comment by gebilaoxiong** [comment]
@yyx990803

Thanks, your code is pretty

**Comment by gebilaoxiong** [comment]
@yyx990803 there is a small problem

https://github.com/vuejs/vue/commit/c3cdfcfa68126783307d74fca22b136cd41c474a#diff-6964744ffd2d3536df0f526fb0e2c286R142

if child is `async placeholder` and it is no tag

can I fix it...

**Pull request closed without merging**